### PR TITLE
App Dev Guide: Environment setup and getting started

### PR DESCRIPTION
**Pull request by ojalatodd**
Old Getting Started tutorial in the App Dev Guide replaced by two new tutorials:
- Environment Setup
- Getting Started with Hyperledger  Sawtooth

### CODE REVIEW

File: docs/source/app_developers_guide/environment_setup.rst [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  validator_1         | [19:03:47.538 INFO     cli] config [path]: key_dir = "/etc/sawtooth/keys"
+  tp_intkey_python_1  | [19:03:47 INFO    core] register attempt: OK
+
+Open a new terminal so we can connect to the client container:
+
+.. code-block:: console
+
+  % docker exec -it compose_client_1 bash
+
+Your environment is ready! Continue on to :doc:`intro_to_sawtooth`
```
Comment (aludvik) [inline docs/source/app_developers_guide/environment_setup.rst]: This is not true. Any work done in this environment will be lost once the container exits. The demo compose file as it stands is only useful for interacting with sawtooth temporarily. In order to use it for app development, some additional things would need to be added such as mounting a host directory into the container.

File: docs/source/app_developers_guide/environment_setup.rst [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+
+Building sawtooth-core
+----------------------
+
+Most of the components of Sawtooth Lake depend partially on code that must
+first be built. This includes C++/swig code and generating protobuf classes for
+each language. Sawtooth Lake also includes SDKs for other languages, including
+Java and JavaScript, which require building. To build the python portion of
+Sawtooth Lake within vagrant, do:
```
Comment (aludvik) [inline docs/source/app_developers_guide/environment_setup.rst]: This tutorial shouldn't assume it is being followed in vagrant.

File: docs/source/app_developers_guide/environment_setup.rst [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+.. code-block:: console
+
+  $ /project/sawtooth-core/bin/build_python
```
Comment (aludvik) [inline docs/source/app_developers_guide/environment_setup.rst]: If you installed sawtooth using apt-get, you do not have the `/project/sawtooth-core` directory or any of the build scripts. This step cannot be performed.

File: docs/source/app_developers_guide/environment_setup.rst [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+This will allow you to run and test Sawtooth Lake components manually. The
+automated tests for python and all other languages rely on docker to ensure
+reproducibility. To run the automated tests for python, first run the
+following:
+
+.. code-block:: console
+
+  $ /project/sawtooth-core/bin/build_all -l python
```
Comment (aludvik) [inline docs/source/app_developers_guide/environment_setup.rst]: See above.

File: docs/source/app_developers_guide/environment_setup.rst [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+This will create docker images for all the python components and run
+`build_python` inside a build container. You can then run the automated tests
+with:
+
+.. code-block:: console
+
+  $ /project/sawtooth-core/bin/run_tests -x java_sdk -x javascript_sdk
```
Comment (aludvik) [inline docs/source/app_developers_guide/environment_setup.rst]: Same as above.

File: docs/source/app_developers_guide/environment_setup.rst [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+If you are not behind a proxy, you can build and test everything Sawtooth Lake
+has to offer with:
+
+.. code-block:: console
+
+  $ /project/sawtooth-core/bin/build_all
+  $ /project/sawtooth-core/bin/run_tests
```
Comment (aludvik) [inline docs/source/app_developers_guide/environment_setup.rst]: Same as above.

**Comment by ojalatodd** [comment]
Also will be rewriting per discussion with @aludvik yesterday.

File: docs/source/app_developers_guide/environment_setup.rst [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+Installation
+------------
+
+Run the following commands from a terminal window:
+
+.. code-block:: console
+
+  $ echo "deb http://repo.sawtooth.me/ubuntu/0.8/stable xenial universe" >> /etc/apt/sources.list
+  $ apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 6B58B1AC10FB5F63
+  $ apt-get update && apt-get install -y sawtooth
```
Comment (ojalatodd) [inline docs/source/app_developers_guide/environment_setup.rst]: Updated

File: docs/source/app_developers_guide/environment_setup.rst (hunk 3 above)
Comment (ojalatodd) [inline docs/source/app_developers_guide/environment_setup.rst]: This whole build section will be going away in a minute!

File: docs/source/app_developers_guide/environment_setup.rst (hunk 1 above)
Comment (ojalatodd) [inline docs/source/app_developers_guide/environment_setup.rst]: Wrote this up as a starting point, with link to Docker documentation.

File: docs/source/app_developers_guide/environment_setup.rst [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+
+
+Building sawtooth-core
+----------------------
+
+Most of the components of Sawtooth Lake depend partially on code that must
+first be built. This includes C++/swig code and generating protobuf classes for
+each language. Sawtooth Lake also includes SDKs for other languages, including
+Java and JavaScript, which require building. To build the python portion of
```
Comment (ojalatodd) [inline docs/source/app_developers_guide/environment_setup.rst]: Removed altogether.

File: docs/source/app_developers_guide/environment_setup.rst (hunk 2 above)
Comment (ojalatodd) [inline docs/source/app_developers_guide/environment_setup.rst]: Removed section.

File: docs/source/app_developers_guide/environment_setup.rst [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+has to offer with:
+
+.. code-block:: console
+
+  $ /project/sawtooth-core/bin/build_all
+  $ /project/sawtooth-core/bin/run_tests
+
+
+Next Steps
+----------
```
Comment (ojalatodd) [inline docs/source/app_developers_guide/environment_setup.rst]: Fixed

File: docs/source/app_developers_guide/environment_setup.rst [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+The following tools are required:
+
+* `Docker Engine <https://docs.docker.com/engine/installation/>`_ (17.03.0-ce
+  or later)
+* `Docker Compose <https://docs.docker.com/compose/install/>`_ (Linux only)
+
+
+Step One: Clone Repository
+--------------------------
```
Comment (aludvik) [inline docs/source/app_developers_guide/environment_setup.rst]: This process seems to be duplicated across a few docs now. Maybe create a story to create a partial doc that contains these instructions and is included as needed?

File: docs/source/app_developers_guide/environment_setup.rst [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+
+.. code-block:: console
+
+  % docker exec -it compose_client_1 bash
+
+Your environment is ready for experimenting with Sawtooth. However, any work
+done in this environment will be lost once the container exits. The demo
+compose file provided is useful as a starting point for the creation of your
+own Docker-based development environment. In order to use it for app
+development, you need to take additional steps, such as mounting a hose
```
Comment (aludvik) [inline docs/source/app_developers_guide/environment_setup.rst]: type "hose" -> "host"

File: docs/source/app_developers_guide/environment_setup.rst (hunk 10 above)
Comment (ojalatodd) [inline docs/source/app_developers_guide/environment_setup.rst]: Yes, I will create a story to store duplicated content in a single location, and insert it inline.

File: docs/source/app_developers_guide/environment_setup.rst (hunk 11 above)
Comment (ojalatodd) [inline docs/source/app_developers_guide/environment_setup.rst]: Fixed

**Comment by ojalatodd** [comment]
NOTE: this guide will be updated with an addition in the near future to allow user/app-developer to access the validator from the host CLI or host network.  @vaporos

**Pull request merged**